### PR TITLE
Added additional timers to kafka source and Hive publisher.

### PR DESCRIPTION
**Pull request by ibuenros**

### CODE REVIEW

**Comment by ibuenros** [comment]
@htran1 can you review? Thanks!

File: gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java [hunk 1]
```diff
@@ -143,7 +161,9 @@ private boolean createDbIfNotExists(IMetaStoreClient client, String dbName) thro
 
       try {
         client.createDatabase(db);
-        log.info("Created database " + dbName);
+        try (Timer.Context context = this.metricContext.timer(CREATE_HIVE_DATABASE).time()) {
```
Comment (htran1) [inline gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java]: Don't we need the create database call in the try?

File: gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java [hunk 2]
```diff
@@ -145,6 +154,8 @@ private void setLimiterReportKeyListToWorkUnits(List<WorkUnit> workUnits, List<S
 
   @Override
   public List<WorkUnit> getWorkunits(SourceState state) {
+    this.metricContext = Instrumented.getMetricContext(state, KafkaSource.class);
+
```
Comment (htran1) [inline gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java]: What about timing isTopicQualified() to get an idea on the cost of the schema fetch?

File: gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java (hunk 1 above)
Comment (ibuenros) [inline gobblin-hive-registration/src/main/java/gobblin/hive/metastore/HiveMetaStoreBasedRegister.java]: Yes, thanks. Fixed.

File: gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java (hunk 2 above)
Comment (ibuenros) [inline gobblin-modules/gobblin-kafka-common/src/main/java/gobblin/source/extractor/extract/kafka/KafkaSource.java]: Done.

**Pull request merged**